l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/CreateNewChain.t.sol

```diff
@@ -0,0 +1,58 @@
+// // SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+import {StateTransitionManagerTest} from "./_StateTransitionManager_Shared.t.sol";
+import {Diamond} from "solpp/state-transition/libraries/Diamond.sol";
+import {DiamondProxy} from "solpp/state-transition/chain-deps/DiamondProxy.sol";
+
+contract createNewChainTest is StateTransitionManagerTest {
+    function testRevertWhenInitialDiamondCutHashMismatch() public {
+        address baseToken = address(0x3030303);
+        address sharedBridge = address(0x4040404);
+        address admin = bridgehub;
+        uint256 chainId = block.chainid;
+        Diamond.DiamondCutData memory initialDiamondCutData = getDiamondCutData(sharedBridge);
+
+        vm.stopPrank();
+        vm.startPrank(bridgehub);
+        vm.expectRevert(bytes("StateTransition: initial cutHash mismatch"));
+
+        chainContractAddress.createNewChain(chainId, baseToken, sharedBridge, admin, abi.encode(initialDiamondCutData));
+    }
+
+    function testRevertWhenCalledNotByBridgehub() public {
+        address baseToken = address(0x3030303);
+        address sharedBridge = address(0x4040404);
+        address admin = bridgehub;
+        uint256 chainId = block.chainid;
+        Diamond.DiamondCutData memory initialDiamondCutData = getDiamondCutData(diamondInit);
+
+        vm.expectRevert(bytes("StateTransition: only bridgehub"));
```

Comment:
It doesn't seem that coverage captured modifier's revert, researching how to make it work

Reply:
done